src/services/__tests__/MatchService.unit.spec.ts

```diff
@@ -13,33 +14,64 @@ beforeEach(() => {
+    });
+    it("updates queue with update player teams", async () => {
+      const mockBallChasers = BallChaserQueueBuilder.many(6, { team: null });
+      jest.mocked(QueueRepository.getAllBallChasersInQueue).mockResolvedValueOnce(mockBallChasers);
+
+      await createRandomMatch();
+      const addMatchMock = jest.mocked(ActiveMatchRepository.addActiveMatch);
 
-  it("updates queue with update player teams", async () => {
-    const mockBallChasers = BallChaserQueueBuilder.many(6, { team: null });
-    jest.mocked(QueueRepository.getAllBallChasersInQueue).mockResolvedValueOnce(mockBallChasers);
+      expect(addMatchMock).toHaveBeenCalled();
+    });
+    it("removes players from queue", async () => {
+      const mockBallChasers = BallChaserQueueBuilder.many(6);
+      jest.mocked(QueueRepository.getAllBallChasersInQueue).mockResolvedValueOnce(mockBallChasers);
 
-    await createRandomMatch();
-    const addMatchMock = jest.mocked(ActiveMatchRepository.addActiveMatch);
+      await createRandomMatch();
+      const removePlayersMock = jest.mocked(QueueRepository.removeAllBallChasersFromQueue);
 
-    expect(addMatchMock).toHaveBeenCalled();
+      expect(removePlayersMock).toHaveBeenCalled();
+    });
   });
+  describe("Captains button was pressed", () => {
+    it("last available queueMember is placed on blue team", async () => {
+      const mockBallChasers: PlayerInQueue[] = [];
+      const mockOrangeBallChasers = BallChaserQueueBuilder.many(3, { team: Team.Orange });
+      const mockBlueBallChasers = BallChaserQueueBuilder.many(2, { team: Team.Blue });
+      const mockBallChaserNoTeam = BallChaserQueueBuilder.single({ team: null, isCap: false });
+      mockBallChasers.push(mockBallChaserNoTeam);
+      mockOrangeBallChasers.forEach((p) => {
+        mockBallChasers.push(p);
+      });
+      mockBlueBallChasers.forEach((p) => {
+        mockBallChasers.push(p);
+      });
+
+      jest.mocked(QueueRepository.getAllBallChasersInQueue).mockResolvedValueOnce(mockBallChasers);
+
+      const players = await createMatchFromChosenTeams();
 
-  it("removes players from queue", async () => {
-    const mockBallChasers = BallChaserQueueBuilder.many(6, { team: null });
-    jest.mocked(QueueRepository.getAllBallChasersInQueue).mockResolvedValueOnce(mockBallChasers);
+      expect(players.filter((player) => player.team === Team.Orange)).toHaveLength(3);
+      expect(players.filter((player) => player.team === Team.Blue)).toHaveLength(3);
+    });
+    it("removes players from queue", async () => {
+      const mockBallChasers = BallChaserQueueBuilder.many(6);
+      jest.mocked(QueueRepository.getAllBallChasersInQueue).mockResolvedValueOnce(mockBallChasers);
 
-    await createRandomMatch();
-    const removePlayersMock = jest.mocked(QueueRepository.removeAllBallChasersFromQueue);
+      await createRandomMatch();
+      const removePlayersMock = jest.mocked(QueueRepository.removeAllBallChasersFromQueue);
```

Comment:
You can just add this assertion to the test above since it's not dependent on anything specific.

Reply:
I have the same test in the random one so I just did the same test for Captains.

Are you wanting me to not have the removal as it's own test then, correct.

Reply:
Yea you can change it for both. The adding of the teams and the removal of the players from the queue occur in the same context for both random and captains, so I think merging them would make sense.

Reply:
Fixed in next commit